dbt/adapters/databricks/connections.py

```diff
@@ -578,10 +580,13 @@ def pollRefreshPipeline(
         stopped_states = ("COMPLETED", "FAILED", "CANCELED")
         host: str = self._creds.host or ""
         headers = self._cursor.connection.thrift_backend._auth_provider._header_factory()
-        headers["User-Agent"] = self._user_agent
 
-        pipeline_id = _get_table_view_pipeline_id(host, headers, model_name)
-        pipeline = _get_pipeline_state(host, headers, pipeline_id)
+        session = Session()
+        session.auth = BearerAuth(headers)
+        session.headers = {"User-Agent": self._user_agent}
```

Comment:
I set the headers on the `Session` now so we don't need to pass `headers` to `_get_pipeline_state` and `_get_update_error_message` etc.